src/Files.App/Utils/RecentItem/RecentItems.cs

```diff
@@ -109,7 +109,8 @@ public async Task UpdateRecentFoldersAsync()
 		/// </summary>
 		public async Task<List<RecentItem>> ListRecentFilesAsync()
 		{
-			return (await Win32Helper.GetShellFolderAsync(QuickAccessGuid, false, true, 0, int.MaxValue)).Enumerate
+			// Since the maximum number of recent files is 20, we set the count to 20 to avoid loading of unnecessary shell items.
+			return (await Win32Helper.GetShellFolderAsync(QuickAccessGuid, false, true, 0, 20)).Enumerate
```

Comment:
This change is not perfect because there are not always 20 recent files. But I don't know how to find out the number of recent files without affecting performance, so I fixed the number of items to 20.

Reply:
I think the number 20 is pretty small, while we might have ways to get count of items in that folder using DirectoryInfo or FindFirstFile/FindNextFile functions.

Reply:
Fair enough